pkg/controller.v1/pytorch/pytorchjob_controller.go

```diff
@@ -171,6 +171,13 @@ func (r *PyTorchJobReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{Requeue: true, RequeueAfter: t}, nil
 	}
 
+	// if we create a pytorchjob, operator will create a podgroup later,if there is no enough resources in cluster, the podgroup will be pending, the reconcile function will break.
+	// if there are enough resources available later, the status of podgroup will be inqueue. then the reconcile function should continue to run. but the operator have not listening the podgroup
+	// len(pytorchjob.Status.Conditions) == 1 means job.status = Created, we should reconcile job a little seconds later
+	if len(pytorchjob.Status.Conditions) == 1 {
```

Comment:
This seems to be a hack to me

Reply:
maybe this condition could be more self-explanatory to reflect the condition mentioned in the comments.

Reply:
I changed the code,  is it a little easier to understand？